File: app/src/main/java/com/example/kolomachinetest/data/ApiResponse.java
package com.example.kolomachinetest.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class ApiResponse implements Parcelable {

    @SerializedName("code")
    public int code;

    @SerializedName("status")
    public String status;

    @SerializedName("copyright")
    public String copyright;

    @SerializedName("attributionText")
    public String attributionText;

    @SerializedName("attributionHTML")
    public String attributionHTML;

    @SerializedName("etag")
    public String etag;

    @SerializedName("data")
    public Data data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getCopyright() {
        return copyright;
    }

    public void setCopyright(String copyright) {
        this.copyright = copyright;
    }

    public String getAttributionText() {
        return attributionText;
    }

    public void setAttributionText(String attributionText) {
        this.attributionText = attributionText;
    }

    public String getAttributionHTML() {
        return attributionHTML;
    }

    public void setAttributionHTML(String attributionHTML) {
        this.attributionHTML = attributionHTML;
    }

    public String getEtag() {
        return etag;
    }

    public void setEtag(String etag) {
        this.etag = etag;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.code);
        dest.writeString(this.status);
        dest.writeString(this.copyright);
        dest.writeString(this.attributionText);
        dest.writeString(this.attributionHTML);
        dest.writeString(this.etag);
        dest.writeParcelable(this.data, flags);
    }

    public void readFromParcel(Parcel source) {
        this.code = source.readInt();
        this.status = source.readString();
        this.copyright = source.readString();
        this.attributionText = source.readString();
        this.attributionHTML = source.readString();
        this.etag = source.readString();
        this.data = source.readParcelable(Data.class.getClassLoader());
    }

    public ApiResponse() {
    }

    protected ApiResponse(Parcel in) {
        this.code = in.readInt();
        this.status = in.readString();
        this.copyright = in.readString();
        this.attributionText = in.readString();
        this.attributionHTML = in.readString();
        this.etag = in.readString();
        this.data = in.readParcelable(Data.class.getClassLoader());
    }

    public static final Parcelable.Creator<ApiResponse> CREATOR = new Parcelable.Creator<ApiResponse>() {
        @Override
        public ApiResponse createFromParcel(Parcel source) {
            return new ApiResponse(source);
        }

        @Override
        public ApiResponse[] newArray(int size) {
            return new ApiResponse[size];
        }
    };
}

File: app/src/main/java/com/example/kolomachinetest/api/repo/marvel/data/ApiResponse.kt
package com.example.kolomachinetest.api.repo.marvel.data

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import android.os.Parcel
import com.example.kolomachinetest.api.repo.marvel.data.ApiResponse

class ApiResponse : Parcelable {
    @SerializedName("code")
    var code = 0

    @SerializedName("status")
    var status: String? = null

    @SerializedName("copyright")
    var copyright: String? = null

    @SerializedName("attributionText")
    var attributionText: String? = null

    @SerializedName("attributionHTML")
    var attributionHTML: String? = null

    @SerializedName("etag")
    var etag: String? = null

    @SerializedName("data")
    var data: Data? = null
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.code)
        dest.writeString(status)
        dest.writeString(copyright)
        dest.writeString(attributionText)
        dest.writeString(attributionHTML)
        dest.writeString(etag)
        dest.writeParcelable(data, flags)
    }

    fun readFromParcel(source: Parcel) {
        this.code = source.readInt()
        status = source.readString()
        copyright = source.readString()
        attributionText = source.readString()
        attributionHTML = source.readString()
        etag = source.readString()
        data = source.readParcelable(Data::class.java.classLoader)
    }

    constructor()
    constructor(`in`: Parcel) {
        this.code = `in`.readInt()
        status = `in`.readString()
        copyright = `in`.readString()
        attributionText = `in`.readString()
        attributionHTML = `in`.readString()
        etag = `in`.readString()
        data = `in`.readParcelable(Data::class.java.classLoader)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<ApiResponse> = object : Parcelable.Creator<ApiResponse> {
            override fun createFromParcel(source: Parcel): ApiResponse {
                return ApiResponse(source)
            }

            override fun newArray(size: Int): Array<ApiResponse?> {
                return arrayOfNulls(size)
            }
        }
    }
}